index.js

```diff
@@ -30,12 +30,14 @@ function compile(options, getData) {
 }
 
 module.exports = function (data, options) {
-	return compile(options, function(file) {
-		if (file.data) {
-			data = _.merge(file.data, data);
-		}
-
-		return data || {};
+	return compile(options, function (file) {
+		return function (data) {
+			if (file.data) {
+				data = _.merge({}, file.data, data);
+			}
+
+			return data || {};
+		}(data);
```

Comment:
Let’s say we are processing two files.
File 1’s gulp-data is `{foo: 'one'}`, file 2’s gulp-data is `{foo: 'two'}` and gulp-template’s data parameter is `{bar: 'bar'}`.

Without the wrapper, the first call to `getData` will do:
`data = _.merge({}, {foo: 'one'}, {bar: 'bar'})`, which sets data to `{foo: 'one', bar: 'bar'}`.
The second call to `getData` will do:
`data = _.merge({}, {foo: 'two'}, {foo: 'one', bar: 'bar'})`, which sets again data to `{foo: 'one', bar: 'bar'}` while we expect `{foo: 'two', bar: 'bar'}`.

The second call should have done:
`data = _.merge({}, {foo: 'two'}, {bar: 'bar'})`.
The function wrapper does this because it is called each time with the original data variable `{bar: 'bar'}`.